src/exceptions/OpenJDK-Assembly-exception-2.0.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<SPDXLicenseCollection xmlns="http://www.spdx.org/license">
+   <exception licenseId="OpenJDK-assembly-exception-2.0" name="Open JDK Assembly exception 2.0">
+      <crossRefs>
+         <crossRef>http://openjdk.java.net/legal/assembly-exception.html
+         </crossRef>
+      </crossRefs>
+      <titleText>OpenJDK Assembly Exception</titleText>
+      <p>
+         The OpenJDK source code made available by Oracle America, Inc.
+         (Oracle) at openjdk.java.net ("OpenJDK Code") is distributed
+         under the terms of the GNU General Public License
+         &lt;http://www.gnu.org/copyleft/gpl.html&gt; version 2 only
+         ("GPL2"), with the following clarification and special
+         exception.
+      </p>
+
+      <list>
+         <item>
+            <p>Linking this OpenJDK Code statically or dynamically with
```

Comment:
I haven't looked at our existing exceptions for precedent here, but to me it feels like the text you've included in the `<list>` element is the legal text of the exception.  The earlier “The OpenJDK source code made available…” paragraph sounds like part of a grant that associates the exception with a particular set of code (much like a GPL header at the beginning of a file associates the GPL license with the code in the file).  The final “As such, it allows licensees and sublicensees…” seems like a combination non-normative notes and a contributor license agreement.

Reply:
The Eclipse OMR and Eclipse OpenJ9 projects both use the OpenJDK assembly exception. I'm not happy with their LICENSE files and will get them to update them.

e.g.

https://github.com/eclipse/openj9/blob/master/jcl/src/openj9.dataaccess/share/classes/com/ibm/dataaccess/ByteArrayMarshaller.java

Reply:
> https://github.com/eclipse/openj9/blob/master/jcl/src/openj9.dataaccess/share/classes/com/ibm/dataaccess/ByteArrayMarshaller.java

The Eclipse OpenJ9 project seems to cover the grant association [in their per-file header][1], which seems like it make's the “The OpenJDK source code made available…” paragraph superfluous for that file.  And they cover the contributor license agreement section with [an explicit CLA and sign-off requirement][2].  So at least some parts of the [upstream exception page][3] seem redundant/misleading in that context.  For OpenJ9 (which is not available under openjdk.java.net?), the exception seems to allow folks to link OpenJ9 against the [designated exception modules][4], but it's the copyright holder (not always Oracle) granting the exception.  A generic form of the exception would be closer to:

> Linking this OpenJDK Code statically or dynamically with other code is making a combined work based on this library.  Thus, the terms and conditions of the GNU General Public License <http://www.gnu.org/copyleft/gpl.html> version 2 only ("GPL2") cover the whole combination.
>
> As a special exception, the copyright holders and contributors give you permission to link this OpenJDK Code with certain code as indicated at http://openjdk.java.net/legal/exception-modules-2007-05-08.html ("Designated Exception Modules") to produce an executable, regardless of the license terms of the Designated Exception Modules, and to copy and distribute the resulting executable under GPL2, provided that the Designated Exception Modules continue to be governed by the licenses under which they were offered by Oracle.

[1]: https://github.com/eclipse/openj9/blob/8fc87280857fe06e0d91409abce26e751c1dc6ae/jcl/src/openj9.dataaccess/share/classes/com/ibm/dataaccess/ByteArrayMarshaller.java#L15-L16
[2]: https://github.com/eclipse/openj9/blob/8fc87280857fe06e0d91409abce26e751c1dc6ae/CONTRIBUTING.md#legal-considerations
[3]: http://openjdk.java.net/legal/assembly-exception.html
[4]: http://openjdk.java.net/legal/exception-modules-2007-05-08.html

Reply:
@waynebeaton  - @wking  has a good point: considering the inbound policy is DCO sign-off (not sure what the need for the ECA is on top of that, but that's another story), all contributors retain their copyright. Thus the exception shouldn't be specific to Oracle. I don't know the history here is and maybe it was appropriate at one time to only come from Oracle if they were the only copyright holder (I believe, they use an assignment agreement for contributions to their project, so this could have worked under that paradigm). But given the Eclipse Foundation licensing policy this exception may need to be updated.